### PR TITLE
check input_device is valid

### PR DESCRIPTION
**Pull request by zhongqiliang**
b/280319229

### CODE REVIEW

File: starboard/android/shared/input_events_generator.cc [hunk 1]
```diff
@@ -800,15 +809,18 @@ bool InputEventsGenerator::CreateInputEventsFromGameActivityEvent(
...
-  UpdateDeviceFlatMapIfNecessary(android_event);
-  ProcessJoyStickEvent(kLeftX, AMOTION_EVENT_AXIS_X, android_event, events);
-  ProcessJoyStickEvent(kLeftY, AMOTION_EVENT_AXIS_Y, android_event, events);
-  ProcessJoyStickEvent(kRightX, AMOTION_EVENT_AXIS_Z, android_event, events);
-  ProcessJoyStickEvent(kRightY, AMOTION_EVENT_AXIS_RZ, android_event, events);
-
-  // Remember the "hat" input values (dpad on the game controller) to help
-  // differentiate hat vs. stick fallback events.
-  UpdateHatValuesAndPossiblySynthesizeKeyEvents(android_event, events);
+  jobject device = GetDevice(android_event);
```
[the remark this answers is not in the document]
Comment (zhongqiliang) [reply starboard/android/shared/input_events_generator.cc]: Updated

**Comment by zhongqiliang** [comment]
> 

From the code here, https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:starboard/android/shared/input_events_generator.cc;l=441?q=p:cobalt%20InputEventsGenerator::ProcessJoyStickEvent
flat is the value from GetFlat(), offset is the value from GameActivityMotionEvent, they are both used for comparison.
"GetFlat() might actually be obsolete", is there any way to confirm this?

Which value is the same as flat?
typedef struct GameActivityMotionEvent {
    int32_t deviceId;
    int32_t source;
    int32_t action;

    int64_t eventTime;
    int64_t downTime;

    int32_t flags;
    int32_t metaState;

    int32_t actionButton;
    int32_t buttonState;
    int32_t classification;
    int32_t edgeFlags;

    uint32_t pointerCount;
    GameActivityPointerAxes
        pointers[GAMEACTIVITY_MAX_NUM_POINTERS_IN_MOTION_EVENT];

    int historySize;
    long* historicalEventTimes;
    float* historicalAxisValues;

    float precisionX;
    float precisionY;
} GameActivityMotionEvent;

**Comment by tphamg** [comment]
GetFlat(... AMOTION_EVENT_AXIS_X) likely maps to GameActivityMotionEvent.pointers[i].axisValues[AMOTION_EVENT_AXIS_X] where i ranges from 0 to GameActivityMotionEvent.pointerCount-1. For joystick events, I'm guessing that pointerCount will only be 1, but maybe if the game controller has a left and right joystick, the pointer count may be > 1 ?

**Comment by zhongqiliang** [comment]
The goal of this change is to fix the crash. I 'd not change Gerry Fan's implementation logic.

**Comment by zhongqiliang** [comment]
I closed the change by mistake, I will merge it when the checks are finished.

**Pull request merged**